worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -96,130 +95,140 @@ class FetchHandler(val conf: CelebornConf, val transportConf: TransportConf)
-          workerSource.startTimer(WorkerSource.OPEN_STREAM_TIME, streamShuffleKey)
-          handleOpenStreamInternal(
-            client,
-            shuffleKey,
-            fileName,
-            startIndex,
-            endIndex,
-            initialCredit,
-            r,
-            false,
-            readLocalShuffle)
-        } catch {
-          case _: Exception =>
-            // process legacy OpenStream RPCs
-            logDebug("Open stream with legacy RPCs")
-            try {
-              val decodedMsg = Message.decode(r.body().nioByteBuffer())
-              val (shuffleKey, fileName) =
-                if (decodedMsg.`type`() == Type.OPEN_STREAM) {
-                  val openStream = decodedMsg.asInstanceOf[OpenStream]
-                  (
-                    new String(openStream.shuffleKey, StandardCharsets.UTF_8),
-                    new String(openStream.fileName, StandardCharsets.UTF_8))
-                } else {
-                  val openStreamWithCredit = decodedMsg.asInstanceOf[OpenStreamWithCredit]
-                  (
-                    new String(openStreamWithCredit.shuffleKey, StandardCharsets.UTF_8),
-                    new String(openStreamWithCredit.fileName, StandardCharsets.UTF_8))
-                }
-              streamShuffleKey = shuffleKey
-              streamFileName = fileName
-              var startIndex = 0
-              var endIndex = 0
-              var initialCredit = 0
-              getRawFileInfo(shuffleKey, fileName).getPartitionType match {
-                case PartitionType.REDUCE =>
-                  startIndex = decodedMsg.asInstanceOf[OpenStream].startMapIndex
-                  endIndex = decodedMsg.asInstanceOf[OpenStream].endMapIndex
-                case PartitionType.MAP =>
-                  initialCredit = decodedMsg.asInstanceOf[OpenStreamWithCredit].initialCredit
-                  startIndex = decodedMsg.asInstanceOf[OpenStreamWithCredit].startIndex
-                  endIndex = decodedMsg.asInstanceOf[OpenStreamWithCredit].endIndex
-                case PartitionType.MAPGROUP =>
-              }
-              handleOpenStreamInternal(
-                client,
-                shuffleKey,
-                fileName,
-                startIndex,
-                endIndex,
-                initialCredit,
-                r,
-                true)
-            } catch {
-              case e: IOException =>
-                handleRpcIOException(client, r.requestId, streamShuffleKey, streamFileName, e)
-            }
-        } finally {
-          r.body().release()
-          workerSource.stopTimer(WorkerSource.OPEN_STREAM_TIME, streamShuffleKey)
-        }
+        handleRpcRequest(client, r)
       case unknown: RequestMessage =>
         throw new IllegalArgumentException(s"Unknown message type id: ${unknown.`type`.id}")
     }
   }
 
+  private def handleRpcRequest(client: TransportClient, rpcRequest: RpcRequest): Unit = {
```

Comment:
I think this change is unnecessary. There is a discussion about  about the code style of this block.

Reply:
Emm...it's to improve the code readability.

Reply:
IMO the refactor is good for readability.